src/mraa.c

```diff
@@ -1130,6 +1140,32 @@ mraa_link_targets(const char* filename, const char* targetname)
     }
 }
 
+mraa_result_t
+mraa_find_uart_bus_pci(const char* pci_dev_path, char** dev_name)
+{
+    char path[PATH_MAX];
+    int max_allowable_len = 16;
+    snprintf(path, PATH_MAX-1, "%s", pci_dev_path);
+    if (!mraa_file_exist(path)) return MRAA_ERROR_UNSPECIFIED;
+
+    struct dirent **namelist;
+    int n = scandir(path, &namelist, NULL, alphasort);
+    if (n <= 0) {
+        syslog(LOG_ERR, "Failed to find expected UART bus");
+        return MRAA_ERROR_NO_RESOURCES;
```

Comment:
1. As we've already checked the directory to exist before, the only reason we get here is error in `scandir()`, but unlikely related to the device/path itself not existing (maybe permissions, etc) - so here something like `INVALID_RESOURCE` or `ERROR_UNSPECIFIED` (as a catchall "things went wrong") is a better fit here;
1. I'd suggest to log the `errno` value to syslog here, to make it easier for user to debug the problem;

Reply:
done.